cmd/run.go

```diff
@@ -407,18 +407,15 @@ a commandline interface for interacting with it.`,
 				case lib.TimeoutError:
 					switch e.Place() {
 					case "setup":
-						logrus.WithField("hint", e.Hint()).Error(err)
-						return ExitCode{errors.New("Setup timeout"), setupTimeoutErrorCode}
+						return ExitCode{err, setupTimeoutErrorCode, e.Hint()}
 					case "teardown":
-						logrus.WithField("hint", e.Hint()).Error(err)
-						return ExitCode{errors.New("Teardown timeout"), teardownTimeoutErrorCode}
+						return ExitCode{err, teardownTimeoutErrorCode, e.Hint()}
 					default:
-						logrus.WithError(err).Error("Engine timeout")
-						return ExitCode{errors.New("Engine timeout"), genericTimeoutErrorCode}
+						return ExitCode{err, genericTimeoutErrorCode, ""}
```

Comment:
I am starting to think that it will be a good idea if we have errors that both have `ErrorCode` and `Hint` maybe as two different interfaces/implmentations. And a `ExitCodeFromError` function that gets those and just generates the `ExitCode` based on the `ErrorCode` in the error and a possible `Hint`. 
At the moment this code is pretty okay but once we start adding more possible `ExitCodeError`s it will grow and it will probably not only be here

Reply:
Our current `ExitCode` does have the underlying error already, not sure why we need more.

Reply:
Well than probably we should extend it to use the provided `error` to determine the `ErrorCode` and `Hint` from it. 
So in this code:
https://github.com/loadimpact/k6/blob/724b72acc84c8b3d4f4101473f3237594a13f6e1/cmd/root.go#L87-L100
we can do something like 
```golang
		code := -1
		var logger logrus.FieldLogger = logrus.StandardLogger()
		if e, ok := err.(ExitCode); ok {
                        if codeErr, ok := e.(GetCoderIntrerface); ok {
			    code = codeErr.GetErrorCode()
                        }
                        if hintErr, ok := e.(GetHintInterface); ok {
				logger = logger.WithField("hint", hintErr.GetHint())
			}
		}
		logger.Error(err)
		os.Exit(code)
```
Obviously the names of the interfaces are terrible :), but I hope this is better explained. 
And then we just make all the errors that we want to make it to ExitCode implement both interfaces or ... not :)

Reply:
@MStoykov I'm not sure it's worth to do it, as we only use `ExitCode` as the entrypoint for reporting meaningful error to user. Any other kind of errors seem to be for internal.

Reply:
Any idea @na-- @imiric